sdk/src/test/scala/com/horizen/consensus/ConsensusDataProviderTest.scala

```diff
@@ -0,0 +1,272 @@
+      acc
+    }
+  }
+
+  @tailrec
+  final def generateBlockIdsAndInfosIter(previousId: ModifierId,
+                                         secondsInSlot: Int,
+                                         nextTimestamp: Long,
+                                         acc: ListBuffer[(ModifierId, SidechainBlockInfo)],
+                                         vrfData: List[Option[(VrfProof, VrfProofHash)]]): Seq[(ModifierId, SidechainBlockInfo)] = {
+    vrfData.headOption match {
+      case Some(Some((vrfProof, vrfProofHash))) => {
+        val idInfo = generateSidechainBlockInfo(previousId, nextTimestamp, vrfProof, vrfProofHash)
+        acc += idInfo
+        generateBlockIdsAndInfosIter(idInfo._1, secondsInSlot, nextTimestamp + secondsInSlot, acc, vrfData.tail)
+      }
+      case Some(None) => generateBlockIdsAndInfosIter(previousId, secondsInSlot, nextTimestamp + secondsInSlot, acc, vrfData.tail)
+      case None => acc
+    }
+  }
+
+  private def generateSidechainBlockInfo(parentId: ModifierId, timestamp: Long, vrfProof: VrfProof, vrfProofHash: VrfProofHash): (ModifierId, SidechainBlockInfo) = {
+    val newBlockId = bytesToId(Utils.doubleSHA256Hash(parentId.getBytes))
+    val blockInfo =
+      new SidechainBlockInfo(0, 0, parentId, timestamp, ModifierSemanticValidity.Valid, Seq(), dummyWithdrawalEpochInfo, vrfProof, vrfProofHash)
+
+    (newBlockId, blockInfo)
+  }
+
+
+  private def slotsPresentationToVrfData(slotsRepresentations: List[List[Int]], prefix: String = ""):  List[List[Option[(VrfProof, VrfProofHash)]]] = {
+    slotsRepresentations.zipWithIndex.map{case (slotsRepresentationsForEpoch, epochIndex) =>
+      slotsRepresentationsForEpoch.zipWithIndex.map{
+        case (1, slotIndex) =>
+          Some(new VrfProof(s"${prefix}proof${epochIndex}${slotIndex}".getBytes), new VrfProofHash(s"${prefix}proofHash${epochIndex}${slotIndex}".getBytes))
+        case (0, _) =>
+          None
+        case _ => throw new IllegalArgumentException
+      }
+    }
+  }
+
+  def getFullConsensusEpochInfoForBlock(idInfo: (ModifierId, SidechainBlockInfo)): FullConsensusEpochInfo = {
+    getFullConsensusEpochInfoForBlock(idInfo._1, idInfo._2)
+  }
+
+  def getInfoForCheckingBlockInEpochNumber(epochNumber: Int): FullConsensusEpochInfo = {
+    getFullConsensusEpochInfoForBlock(blockIdAndInfosPerEpoch(epochNumber - 1).head)
+  }
+}
+
+class BlocksInfoProvider extends SidechainBlockInfoProvider {
+  private val storage = mutable.Map[ModifierId, SidechainBlockInfo]()
+
+  override def blockInfoById(blockId: ModifierId): SidechainBlockInfo = storage(blockId)
+  def addBlockInfo(blockId: ModifierId, sidechainBlockInfo: SidechainBlockInfo): Unit = storage.put(blockId, sidechainBlockInfo)
+}
+
+class ConsensusDataProviderTest extends CompanionsFixture{
+  val generator: SidechainBlockFixture = new SidechainBlockFixture {}
+  val dummyWithdrawalEpochInfo = utils.WithdrawalEpochInfo(0, 0)
+
+
+  @Test
+  def test(): Unit = {
+    val slotsInEpoch = 10
```

Comment:
In Ouroboros paper they told, that we need to have 24k of slots in epoch to be able to divide it by 3. So, the number of slots should be modulus 3.
Do we need to check `slotsInEpoch` value in `NetworkParams` or somewhere else?
The same here in current test - set `slotsInEpoch = 9`, for example.